stumpy/cache.py

```diff
@@ -117,3 +145,72 @@ def _get_cache():
     site_pkg_dir = site.getsitepackages()[0]
     numba_cache_dir = site_pkg_dir + "/stumpy/__pycache__"
     return [f.name for f in pathlib.Path(numba_cache_dir).glob("*nb*") if f.is_file()]
+
+
+def _recompile():
+    """
+    Recompile all njit functions
+
+    Parameters
+    ----------
+    None
+
+    Returns
+    -------
+    None
+
+    Notes
+    -----
+    If the `numba` cache is enabled, this results in saving (and/or overwriting)
+    the cached numba functions to disk.
+    """
+    warnings.warn(CACHE_WARNING)
+    for module_name, func_name in get_njit_funcs():
+        module = importlib.import_module(f".{module_name}", package="stumpy")
+        func = getattr(module, func_name)
+        func.recompile()
+
+    return
+
+
+def _save():
+    """
+    Save all njit functions
+
+    Parameters
+    ----------
+    None
+
+    Returns
+    -------
+    None
+    """
+    warnings.filterwarnings("once")
+    warnings.warn(CACHE_WARNING)
+    _enable()
+    _recompile()
+    warnings.filterwarnings("default")
+
+    return
+
+
+def save():
+    """
+    Save/overwrite all the cache data files of
+    all-so-far compiled njit functions.
+
+    Parameters
+    ----------
+    None
+
+    Returns
+    -------
+    None
+    """
```

Comment:
@seanlaw
Now that we are creating wrapper around _save(), I was wondering if we should add warnings.warn(CACHE_WARNING) here, and remove this warning in _save(), _enable(), and _recompile(). Because, IIUC, the goal is to just warn the user and not the developer.

Reply:
Yes! I was thinking the same thing. Please do as you said